indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -851,11 +1028,58 @@ public TaskStatusPlus apply(TaskRunnerWorkItem workItem)
           )
       );
     };
+    Collection<? extends TaskRunnerWorkItem> dataSourceFilter = collectionToFilter;
+    if (dataSource != null) {
+      dataSourceFilter = collectionToFilter
+          .stream()
+          .filter(task -> task.getDataSource().equals(dataSource))
+          .collect(Collectors.toList());
+    }
+    return Lists.newArrayList(
+        AuthorizationUtils.filterAuthorizedResources(
+            req,
+            dataSourceFilter,
+            raGenerator,
+            authorizerMapper
+        )
+    );
+  }
 
+  private List<TaskStatusPlus> securedTaskStatusPlus(
+      List<TaskStatusPlus> collectionToFilter, String dataSource,
+      HttpServletRequest req
+  )
+  {
+    Function<TaskStatusPlus, Iterable<ResourceAction>> raGenerator = taskStatusPlus -> {
+      final String taskId = taskStatusPlus.getId();
+      //final Optional<Task> optionalTask = taskStorageQueryAdapter.getTask(taskId);
+      TaskInfo taskInfo = taskInfoCache.get(taskId);
```

Comment:
Based on the comments above for `taskInfoCache`, you'll probably want to rethink how this works.

Maybe it would make sense to pass in a map of task id -> datasource (presumably freshly retrieved) to this `securedTaskStatusPlus` method?